### PR TITLE
Module 7 home task has been completed.

### PR DESCRIPTION
**Pull request by Adenmin**

### CODE REVIEW

File: module5/NewsFeedStatistic.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    def __init__(self, newsfeed_file_name):
+        self.newsfeed_file_name = newsfeed_file_name
+        self.text = self.__read_newsfeed()
+        self.letters = self.__get_letters()
+
+    def write_word_statistic(self, csv_file_name) -> None:
+        with open(csv_file_name, 'w') as csvfile:
+            writer = csv.writer(csvfile, delimiter='-')
+            for word, count in self.__get_world_statistic().items():
```
Comment (romkabauer) [inline module5/NewsFeedStatistic.py:16]: I think items is working not properly on arabic keys... It creates corteges with values as part of cortege with index = 0 and key as part of cortege with index = 1. That is why csv file will return bad data... I think we need to handle this situation somehow

Comment (romkabauer) [inline module5/NewsFeedStatistic.py:16]: I think items is working not properly on arabic keys... It creates corteges with values as part of cortege with index = 0 and key as part of cortege with index = 1. That is why csv file will return bad data... I think we need to handle this situation somehow

File: module5/NewsFeedStatistic.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                writer.writerow([word, count])
+
+    def write_letter_statistic(self, csv_file_name) -> None:
+        with open(csv_file_name, 'w') as csvfile:
+            headers = ['letter', 'count_all', 'count_uppercase', 'percentage']
+            writer = csv.DictWriter(csvfile, fieldnames=headers, quoting=csv.QUOTE_NONNUMERIC)
+            writer.writeheader()
+            for letter in set(self.letters.lower()):
+                writer.writerow({'letter': letter,
```
Comment (romkabauer) [inline module5/NewsFeedStatistic.py:25]: I think this function is also work with arabic not properly.. Please check how we can handle it properly.

**Review by romkabauer — CHANGES_REQUESTED** [review]

File: module5/NewsFeedStatistic.py (hunk 1 above)
Comment (Adenmin) [reply module5/NewsFeedStatistic.py:16]: This looks like a display bug. 
If we create csv file by csv.writer and read this file by csv.reader or csv.DictReader, then everything will work correctly.

File: module5/NewsFeedStatistic.py (hunk 2 above)
Comment (Adenmin) [reply module5/NewsFeedStatistic.py:25]: Additional plugins for PyCharm also display the data in the file correctly.
image

File: module5/NewsFeedStatistic.py (hunk 1 above)
Comment (romkabauer) [reply module5/NewsFeedStatistic.py:16]: Yes, it's a feature of readers, we need to specify that data sample have arabic characters to read it properly. So code is working as expected. Will close thread.

File: module5/NewsFeedStatistic.py (hunk 2 above)
Comment (romkabauer) [reply module5/NewsFeedStatistic.py:25]: Yes, it's a feature of readers, we need to specify that data sample have arabic characters to read it properly. So code is working as expected. Will close thread.

**Pull request merged**